### DISCUSSION TITLE
Information about S3 buckets

### DISCUSSION THREAD

**Discussion by XDavidT** [NetApp/harvest#1749] [Ideas]
On our new NetApp, we saw that the S3 object store is now supported, and we started using it for our needs.
We have around 5 buckets currently with high usage.
I want to suggest a metric that shows the following:
 1. Buckets names, quotes, and usage
 2. encrypted/not encrypted
 3. how many keys does each one has
 4. HTTP requests and responses
 5. HTTP server health
 6. files added to a bucket during day/ or any time range

**Comment by rahulguptajss** [comment]
@XDavidT Harvest already ships with s3 metrics from release 22.11.0. There are templates available 
Config :           https://github.com/NetApp/harvest/blob/main/conf/rest/9.7.0/ontap_s3.yaml
Performance : https://github.com/NetApp/harvest/blob/main/conf/zapiperf/cdot/9.8.0/ontap_s3.yaml

**Comment by rahulguptajss** [comment]
Information about these counters is available [here](https://netapp.github.io/harvest/23.02/ontap-metrics/#ontaps3_abort_multipart_upload_failed)

**Reply by XDavidT** [reply]
I've reviewed them, and not all of them are available on my side, such as `ontaps3_size` 
More than that, I saw all of them provide information about requests of HTTP server - latency, requests, rate, and more.. Are there any metrics that provide information about the current status? Main goals is to know:

- How many buckets are available
- Size of each bucket
- count objects on the bucket
- What keys been used at most

**Reply by rahulguptajss** [reply]
We have some of these available as below

- How many buckets are available
        Prometheus Query Number of buckets per cluster: `count(ontaps3_labels) by (cluster)`
        
- Size of each bucket
       Metrics:  https://netapp.github.io/harvest/latest/ontap-metrics/#ontaps3_size
                       https://netapp.github.io/harvest/latest/ontap-metrics/#ontaps3_size
                       
- count objects on the bucket
        This information is available via private cli. We'll add it via #1759.
        
- What keys been used at most
        I don't see it in Rest response. We'll check and get back on this.

**Comment by XDavidT** [comment]
Is there any dashboard specific to viewing those parameters, or is it's only metric we can use to build our own?

**Comment by rahulguptajss** [comment]
@XDavidT There are no dashboard available for S3 metrics currently. We'll need to build one. Could you open a feature request for the same.

**Reply by XDavidT** [reply]
https://github.com/NetApp/harvest/issues/1768
